pkg/services/ngalert/models/alert_rule.go

```diff
@@ -514,6 +516,13 @@ func (alertRule *AlertRule) ValidateAlertRule(cfg setting.UnifiedAlertingSetting
 		}
 	}
 
+	if alertRule.Record != "" {
+		return fmt.Errorf("%w: Storing recording rules is not yet allowed.", ErrAlertRuleFailedValidation)
+	}
```

Comment:
Not really reachable since we do not have API layer fields for this, just making things apparent. This was useful to catch serialization bugs in tests.